tomcat-jta/src/main/java/org/jboss/narayana/tomcat/jta/TransactionalDataSourceFactory.java

```diff
@@ -163,6 +166,9 @@ public Object getObjectInstance(Object obj, Name name, Context context, Hashtabl
 
             mds.setTransactionManager(transactionManager);
             mds.setXaDataSourceInstance(xaDataSource);
+            if (tsr != null) {
+                mds.setTransactionSynchronizationRegistry(tsr);
```

Comment:
Well, you can set it even if null, BasicManagedDataSource doesn't care.

Reply:
OK, I will update it.